### PR TITLE
Minor bug fixes

### PR DESCRIPTION
**Pull request by taylorbell57**
This PR has some minor bug fixes. Specifically, it fixes some small bugs with S3 photometry code, and it fixed a bug where setting fill_value to 'boxcar' in Stage 4 still resulted in points being masked while they should've been left unmasked and simply replaced with the value from the boxcar smoothed data.

### CODE REVIEW

**Comment by taylorbell57** [comment]
Sigh, well I removed some of GitHub's issues, but now we're back to some gcc error that I don't get locally (and I do remember seeing with some recent PRs until the new numpy blas errors starting popping up even more recently). I'll try to read through the error messages closely and see if I can figure out what the issue is

File: environment.yml [hunk 1]
```diff
@@ -4,40 +4,46 @@ channels:
...
+    - george # Needed for GP
+    - h5py
+    - ipykernel
     - ipython
+    - jupyter
     - lmfit
     - matplotlib
     - nbsphinx # Needed for documentation
-    - numpy[version='>=1.20.0']
+    - numpy[version='>=1.20.0,<=1.23'] # Upper limit needed for Apple silicon
```
Comment (kevin218) [inline environment.yml:25]: I see that the upper limit is 1.23 here and 1.22 in [environment_pymc3.yml](https://github.com/kevin218/Eureka/pull/558/files#diff-15acc9e6ddc11f0b6781d4f3a13a02fe1aa14a2aeb776a107b611f63ab4a3c95).  What's the reason for this difference?  Would it just be better to remain consistent and adopt 1.22 here as well or should we aim for maximum flexibility on the version numbers?

**Comment by kevin218** [comment]
I see that Pytest Testing fails with a new error, possibly relating to gcc.

**Comment by taylorbell57** [comment]
Yeah, I'm actively working on that gcc issue - I'll re-request your review once I've figured out how to get gcc working again (my best guess is that a gcc version number changed and broke GitHub's testing)

**Comment by taylorbell57** [comment]
FYI, I had no luck with changing the gcc version, so I reverted those commits and my next best guess is that a cython version is to blame. I'll have to come back to that later (perhaps in a few days). In the meantime, you could approve this PR if you want to and leave the resolution to the GitHub-exclusive gcc error for another PR

**Review by kevin218 — APPROVED** [review]

**Pull request merged**